src/runtime/crt/contrib/stm32/ai_platform.h

```diff
@@ -0,0 +1,383 @@
+ */
+typedef struct ai_intq_info_ {
+  INTQ_CONST ai_float* scale;
+  INTQ_CONST ai_handle zeropoint;
+} ai_intq_info;
+
+/*!
+ * @struct ai_intq_info_list
+ * @ingroup ai_platform
+ * @brief list reporting meta info for quantized networks integer support
+ * when size > 1 it means a per channel out quantization
+ */
+typedef struct ai_intq_info_list_ {
+  ai_u16 flags;                  /*!< optional flags to store intq info attributes */
+  ai_u16 size;                   /*!< number of elements in the the intq_info list  */
+  INTQ_CONST ai_intq_info* info; /*!< pointer to an array of quant info */
+} ai_intq_info_list;
+
+/******************************************************************************/
+/*!
+ * @struct ai_buffer_meta_info
+ * @ingroup ai_platform
+ * @brief Optional meta attributes associated with the I/O buffer.
+ * This datastruct is used also for network querying, where the data field may
+ * may be NULL.
+ */
+typedef struct ai_buffer_meta_info_ {
+  ai_u32 flags;                 /*!< meta info flags */
+  ai_intq_info_list* intq_info; /*!< meta info related to integer format */
+} ai_buffer_meta_info;
+
+/*!
+ * @struct ai_buffer
+ * @ingroup ai_platform
+ * @brief Memory buffer storing data (optional) with a shape, size and type.
+ * This datastruct is used also for network querying, where the data field may
+ * may be NULL.
+ */
+typedef struct ai_buffer_ {
+  ai_buffer_format format;        /*!< buffer format */
+  ai_u16 n_batches;               /*!< number of batches in the buffer */
+  ai_u16 height;                  /*!< buffer height dimension */
+  ai_u16 width;                   /*!< buffer width dimension */
+  ai_u32 channels;                /*!< buffer number of channels */
+  ai_handle data;                 /*!< pointer to buffer data */
+  ai_buffer_meta_info* meta_info; /*!< pointer to buffer metadata info */
+} ai_buffer;
+
+/*!
+ * @struct ai_platform_version
+ * @ingroup ai_platform
+ * @brief Datastruct storing platform version info
+ */
+typedef struct ai_platform_version_ {
+  ai_u8 major;
+  ai_u8 minor;
+  ai_u8 micro;
+  ai_u8 reserved;
+} ai_platform_version;
+
+/*!
+ * @struct ai_network_report
+ * @ingroup ai_platform
+ *
+ * Datastructure to query a network report with some relevant network detail.
+ */
+typedef struct ai_network_report_ {
+  const char* model_name;
+  const char* model_signature;
+  const char* model_datetime;
+
+  const char* compile_datetime;
+
+  const char* runtime_revision;
+  ai_platform_version runtime_version;
+
+  const char* tool_revision;
+  ai_platform_version tool_version;
+  ai_platform_version tool_api_version;
+
+  ai_platform_version api_version;
+  ai_platform_version interface_api_version;
+
+  ai_u32 n_macc;
+
+  ai_u16 n_inputs;
+  ai_u16 n_outputs;
+  ai_buffer* inputs;
+  ai_buffer* outputs;
+
+  ai_buffer activations;
+  ai_buffer params;
+
+  ai_u32 n_nodes;
+
+  ai_signature signature;
+} ai_network_report;
```

Comment:
i'm not sure i see this struct used at all in the codebase. aside from a couple typedefs used in `ai_runtime_api.h`, i'm not sure this file is used. can we either delete it or pare it down considerably? i'm not sure if maybe this file is typically provided by X-Cube, but it would be great to provide a minimal bridge between TVM and X-Cube by just including the typedefs needed for `ai_runtime_api.h`.